src/V3EmitC.cpp

```diff
@@ -2268,6 +2268,18 @@ void EmitCStmts::displayArg(AstNode* dispp, AstNode** elistp, bool isScan, const
             emitDispState.pushArg(' ', nullptr, cvtToStr(argp->widthMin()));
         }
         emitDispState.pushArg(fmtLetter, argp, "");
+        if (fmtLetter == 't' || fmtLetter == '^') {
+            AstSFormatF* fmtp = nullptr;
+            if (AstDisplay* nodep = VN_CAST(dispp, Display))
+                fmtp = nodep->fmtp();
+            else if (AstSFormat* nodep = VN_CAST(dispp, SFormat))
+                fmtp = nodep->fmtp();
+            else
+                fmtp = VN_CAST(dispp, SFormatF);
+            UASSERT_OBJ(fmtp, dispp, "must be either AstDisplay, AstSFormat, or AstSFormatF");
```

Comment:
```suggestion
            UASSERT_OBJ(fmtp, dispp, "Use of %t must be under AstDisplay, AstSFormat, or AstSFormatF");
```

But perhaps it's better instead to have V3Width insert this argument in a place similar to the code you removed (not sure)?